### PR TITLE
Facilitate running the benchmark

### PR DESCRIPTION
**Pull request by cifkao**
- Adding a `--test_only` flag, making it possible to evaluate on the test set once the model is trained.
- Saving the test results in a JSON file, which is more practical than having to check the log file or the TensorBoard summary.

### CODE REVIEW

**Review by ppham27 — CHANGES_REQUESTED** [review]
Can you update the open calls with the `tf.io.gfile` API? https://www.tensorflow.org/api_docs/python/tf/io/gfile

**Comment by cifkao** [comment]
> Can you update the open calls with the `tf.io.gfile` API? https://www.tensorflow.org/api_docs/python/tf/io/gfile

Done!

**Comment by BalloutAI** [comment]
Any idea on Issue #15. I am getting low accuracy when running test_only, even though in training the validation is high.
